src/skin/qml/qmleffectmanifestparametersmodel.cpp

```diff
@@ -0,0 +1,73 @@
+#include "skin/qml/qmleffectmanifestparametersmodel.h"
+
+#include <QModelIndex>
+
+#include "effects/effectmanifest.h"
+
+namespace mixxx {
+namespace skin {
+namespace qml {
+namespace {
+const QHash<int, QByteArray> kRoleNames = {
+        {QmlEffectManifestParametersModel::IdRole, "id"},
+        {QmlEffectManifestParametersModel::NameRole, "name"},
+        {QmlEffectManifestParametersModel::ShortNameRole, "shortName"},
+        {QmlEffectManifestParametersModel::DescriptionRole, "description"},
+        {QmlEffectManifestParametersModel::ControlHintRole, "controlHint"},
+};
+}
+
+QmlEffectManifestParametersModel::QmlEffectManifestParametersModel(
+        EffectManifestPointer pEffectManifest,
+        QObject* parent)
+        : QAbstractListModel(parent), m_pEffectManifest(pEffectManifest) {
+}
+
+QVariant QmlEffectManifestParametersModel::data(const QModelIndex& index, int role) const {
+    const QList<EffectManifestParameterPointer>& parameters = m_pEffectManifest->parameters();
+    if (index.row() >= parameters.size()) {
+        return QVariant();
+    }
+
+    const EffectManifestParameterPointer pParameter = parameters.at(index.row());
+    switch (role) {
+    case QmlEffectManifestParametersModel::IdRole:
+        return pParameter->id();
+    case QmlEffectManifestParametersModel::NameRole:
+        return pParameter->name();
+    case QmlEffectManifestParametersModel::ShortNameRole:
+        return pParameter->shortName();
+    case QmlEffectManifestParametersModel::DescriptionRole:
+        return pParameter->description();
+    case QmlEffectManifestParametersModel::ControlHintRole:
+        return static_cast<int>(pParameter->controlHint());
```

Comment:
Can't we expose the enum directly?

Reply:
Unsure. In the end this would be better, but the enum doesn't conform to our naming standard anyway. I would change that, but I really don't want to cause more conflicts with #2618. I can add a TODO comment to take care of that after that PR has been merged.